src/utils/index.ts

```diff
@@ -27,22 +27,28 @@ function filterArgs(
 // Add all required search params with user inputs as values
 export function formatUrl(
   url: string,
-  params?: string[],
-  args?: Inputs,
+  paramKeys?: string[],
+  paramInputs?: Inputs,
   slug?: Inputs,
-  optionalParams?: Inputs,
+  optionalParamKeys?: string[],
+  optionalParamInputs?: Inputs,
```

Comment:
renamed params, cause I got lost after a while..  would be nice to use an object, but maybe too late for that 🤷

Reply:
I won't impact nuxt scripts. But it may be a breaking change for the PHP and nextJS integration if they use it

Reply:
Once merged, we'll publish a major version and indicate a breaking change.
I think either way there'll be a breaking change

Reply:
I think an array works fine here. Could you clarify in what sense you mean an object would help?

Reply:
oh just instead of using: 

```
export function formatUrl(
  url: string,
  params?: string[],
  args?: Inputs,
  slug?: Inputs,
  optionalParams?: Inputs,
) {..}
```

maybe this instead:

```
export function formatUrl({
  url,
  params,
  args,
  slug,
  optionalParams,
}:FormatUrlParams) {...}
```

Reply:
Ah ok. Yeah, probably not critical enough to introduce a breaking change here. Personally I'm fine with the 5 parameters.